envisage/plugin.py

```diff
@@ -289,7 +289,13 @@ def unregister_services(self):
         service_ids.reverse()
 
         for service_id in service_ids:
-            self.application.unregister_service(service_id)
+            # note the service may have already been individually unregistered
```

Comment:
Nit: This comment may be better placed in the `except ValueError` block. It directly explains why we are excusing the exception. A service can be unregistered by others and the plugin won't know about it. The ValueError indicates it is already unregistered, and the plugin is cool about it.